### PR TITLE
refactor: Remove Cypress `cy.visitWithLogin()` account mocking, improve DBaaS mocks

### PR DESCRIPTION
**Pull request by jdamore-linode**
## Description 📝
This is related to PR #9715 from yesterday -- we got the tests passing by mocking the `databases` feature flag to `true`, when the root cause of the failures was actually that the API `/account` request was getting mocked, and the mocked response did not include "Managed Databases" in the list of capabilities.

This PR makes a handful of changes to address this and restore the DBaaS tests to a state that's more useful for their team.

## Major Changes 🔄
- Adds `Cloud Firewall`, `Vlans`, `Machine Images`, and `Managed Databases` to the Account factory's `capabilities` property
- Removes implicit account mocking behavior from `cy.visitWithLogin()` -- this was added a while ago to get around performance issues with the API `/account` endpoint. I think the performance issues with that endpoint have been improved if not resolved, and opted to remove this behavior altogether due to the confusion caused
- Explicitly mock the account request in the DBaaS tests to ensure that the tests are running with the `Managed Databases` capability available, regardless of the state of the test accounts
- Remove the mocks related to the `databases` feature flag so that the DBaaS team can run the tests against environments where the flag is and is not enabled

## How to test 🧪
We can rely on automated tests to ensure that everything continues running and passing as expected. I want to pay particular attention to the performance of these test runs to ensure that removing the implicit account mocking didn't impact performance in a significant way.

### CODE REVIEW

File: packages/manager/src/factories/account.ts [hunk 1]
```diff
@@ -41,7 +41,11 @@ export const accountFactory = Factory.Sync.makeFactory<Account>({
     'Block Storage',
     'Object Storage',
     'Kubernetes',
+    'Cloud Firewall',
+    'Vlans',
     'LKE HA Control Planes',
+    'Machine Images',
+    'Managed Databases',
```
Comment (jdamore-linode) [inline packages/manager/src/factories/account.ts:48]: I signed in to my personal account and added the capabilities which were present there but missing here.

**Comment by rvenugopal11** [comment]
Hi @jdamore-linode Can you please add Surbhi to these PRs going forward if they affect databases?

**Comment by jdamore-linode** [comment]
> Hi @jdamore-linode Can you please add Surbhi to these PRs going forward if they affect databases?

Hey @rvenugopal11. I'm not able to add Surbhi as a reviewer, I believe because she's not a member of the organization in GitHub. I sent her a link to the PR in Slack when I opened the PR and requested her review there instead.

**Review by abailly-akamai — APPROVED** [review]
Looks good:

- Ran tests locally and they pass as usual ✅ 
- Did not notice any performance degradation from removing account mocking ✅

**Comment by abailly-akamai** [comment]
@jdamore-linode Should I assume the /linode-storage.spec.ts failure is unrelated?

**Review by bnussman-akamai — APPROVED** [review]

**Comment by jdamore-linode** [comment]
> @jdamore-linode Should I assume the `/linode-storage.spec.ts` failure is unrelated?

Yeah, I believe that's the case. I'm going to try rerunning this until I can get a green check, but I'm trying to get a handle on all of these flaky tests too.

**Comment by jdamore-linode** [comment]
Pushed a quick addition that should get the tests passing again in light of the LD feature/API changes made today

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**